src/renderer/state.ts

```diff
@@ -530,19 +577,55 @@ export class AppState {
    */
   public async downloadVersion(ver: RunnableVersion) {
     const { source, state, version } = ver;
+    const {
+      electronMirror,
+      electronNightlyMirror,
+    } = this.electronMirror.sources[this.electronMirror.sourceType];
 
     const isRemote = source === VersionSource.remote;
+    const isDownloaded = state === VersionState.downloaded;
     const isReady = state === VersionState.ready;
     if (!isRemote || isReady) {
       console.log(`State: Already have version ${version}; not downloading.`);
       return;
     }
 
+    if (isDownloaded) {
+      // The electron zip needs to be unzipped as well
+      await this.installer.install(version);
+      return;
+    }
+
     console.log(`State: Downloading Electron ${version}`);
-    await setupBinary(
-      ver,
-      this.electronMirror.sources[this.electronMirror.sourceType],
-    );
+    // Fix the Checksum error
```

Comment:
Is this comment meant to be a TODO? I don't see the context here.

Reply:
Oh yeah! This is to be removed :)